[StaticRuntime] Fix a bug that memory planner ignores subblocks (#146728)

coufon opened the pull request:
Summary:

When Static Runtime graph node has sub-blocks, the memory planner does not consider sub-blocks' inputs as a node's input in memory planner. As the result, such nodes' inputs' lifetime is incorrect and corresponding tensor memory is released earlier than required and causes errors.

Differential Revision: D69195886




cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

swolchok reviewed (approved):
please fix lint

not sure if my approval is actually good enough to land PyTorch PRs or not, but here you go

coufon committed:
[StaticRuntime] Fix a bug that memory planner ignores subblocks (#146855)

Summary:

When Static Runtime graph node has sub-blocks, the memory planner does not consider sub-blocks' inputs as a node's input in memory planner. As the result, such nodes' inputs' lifetime is incorrect and corresponding tensor memory is released earlier than required and causes errors.

Differential Revision: D69195886

coufon commented:
> please fix lint
> 
> not sure if my approval is actually good enough to land PyTorch PRs or not, but here you go

Fixed the lint, thank you!

jeanschmidt commented:
@pytorchbot merge -f "Rocm runners are preventing merges"
